airbyte-integrations/bases/source-acceptance-test/unit_tests/test_json_schema_helper.py

```diff
@@ -208,3 +208,52 @@ def test_get_object_strucutre(object, pathes):
 )
 def test_get_expected_schema_structure(schema, pathes):
     assert get_expected_schema_structure(schema) == pathes
+
+
+@pytest.mark.parametrize(
+    "keys, num_paths, last_value",
+    [
+        (["description"], 1, "Tests that keys can be found inside lists of dicts"),
+        (["option1"], 2, {"a_key": "a_value"}),
+        (["option2"], 1, ["value1", "value2"]),
+        (["nonexistent_key"], 0, None),
```

Comment:
Does having array indexes as keys make sense?  Perhaps they should be excluded. 
You have this in your example as well:
```
After:

schema_helper.find_nodes(keys=["option1"])

[['properties', 'credentials', 'oneOf', 0, 'properties', 'option1']]
```

perhaps this should be the result:
```
[['properties', 'credentials', 'oneOf', 'properties', 'option1']] // no 0
```

This is not a blocking question/suggestion.

Reply:
The previous solution which preserved the whole object/array did allow us to understand depth.  Another option could be something like:

```
[['properties', 'properties/credentials', 'oneOf/properties/option1']]
```

Reply:
So the issue here is that `0` is the indicator for the first element of the list, so there could be something totally different at 
```
[['properties', 'credentials', 'oneOf', 1, 'properties', 'option1']]
```
, or in fact no `'properties', 'option1']` down that path as well. 

I'm intrigued by the slash idea, but still not sure it gets past the "which item in the list is it" question. I.e. 
`'oneOf/properties/option1'` doesn't tell me which `oneOf` selection that `properties/option1` path lives under.

Reply:
To be clear, these aren't list of keys, they're path segments. so option1 is located at `.properties.credentials.oneOf.0.properties.option1` 
or in python 
`schema['properties'[credentials']['oneOf'][0][properties][option1]`. 

The latter is actually how it's calculated with `get_node`, which is why you got the original error when it tried to pull
`schema['properties'[credentials']['oneOf'][{some_dict_here}`
instead of 
`schema['properties'[credentials']['oneOf'][0]`

If there are multiple paths, we get a list of lists of path segments from this function.

Reply:
Good point! The `/` option would be something like `oneOf/0/properties/option1` then, if you wanted to go that route.  Just an idea, not a requirement!

Reply:
Cool, I mostly just wanted to align on the fact that those indices are necessary before moving forward!